airbyte-config/specs/src/main/java/io/airbyte/config/specs/SeedConnectorSpecGenerator.java

```diff
@@ -0,0 +1,133 @@
+import io.airbyte.commons.io.IOs;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.util.MoreIterators;
+import io.airbyte.commons.yaml.Yamls;
+import io.airbyte.config.DockerImageSpec;
+import io.airbyte.config.EnvConfigs;
+import io.airbyte.protocol.models.ConnectorSpecification;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This script is responsible for ensuring that up-to-date {@link ConnectorSpecification}s for every
+ * connector definition in the seed are stored in a corresponding resource file, for the purpose of
+ * seeding the specs into the config database on server startup.
+ * <p>
+ * Specs are stored in a separate file from the definitions in an effort to keep the definitions
+ * yaml files human-readable and easily-editable, as specs can be rather large.
+ */
+public class SeedConnectorSpecGenerator {
+
+  private static final String DOCKER_REPOSITORY_FIELD = "dockerRepository";
+  private static final String DOCKER_IMAGE_TAG_FIELD = "dockerImageTag";
+  private static final String DOCKER_IMAGE_FIELD = "dockerImage";
+  private static final String SPEC_FIELD = "spec";
+  private static final String SPEC_BUCKET_NAME = new EnvConfigs().getSpecCacheBucket();
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SeedConnectorSpecGenerator.class);
+
+  private static final Options OPTIONS = new Options();
+  private static final Option SEED_ROOT_OPTION = new Option("s", "seed-root", true, "path to where seed resource files are stored");
+
+  private final GcsBucketSpecFetcher bucketSpecFetcher;
+
+  SeedConnectorSpecGenerator(final GcsBucketSpecFetcher bucketSpecFetcher) {
+    this.bucketSpecFetcher = bucketSpecFetcher;
+  }
+
+  static {
+    SEED_ROOT_OPTION.setRequired(true);
+    OPTIONS.addOption(SEED_ROOT_OPTION);
+  }
+
+  public static void main(final String[] args) throws Exception {
+    final CommandLine parsed = parse(args);
+    final Path outputRoot = Path.of(parsed.getOptionValue(SEED_ROOT_OPTION.getOpt()));
+
+    final GcsBucketSpecFetcher bucketSpecFetcher = new GcsBucketSpecFetcher(StorageOptions.getDefaultInstance().getService(), SPEC_BUCKET_NAME);
+    final SeedConnectorSpecGenerator seedConnectorSpecGenerator = new SeedConnectorSpecGenerator(bucketSpecFetcher);
+    seedConnectorSpecGenerator.run(outputRoot, ConnectorType.SOURCE);
+    seedConnectorSpecGenerator.run(outputRoot, ConnectorType.DESTINATION);
+  }
+
+  public void run(final Path seedRoot, final ConnectorType connectorType) throws IOException {
```

Comment:
the enum is a nice touch.

Reply:
Unfortunately, I just realized that I probably need to get rid of the enum. Because we want to reuse this class in the cloud build, and the seed definition files in cloud have slightly different names (`source_definitions_mask.yaml` vs `source_definitions.yaml`), so with the current code I am unable to pass the desired filenames to this class from cloud. And I'm guessing we don't want to have something like a `CLOUD_SOURCE` value in this enum, as it is in OSS

Reply:
Actually, I'll probably keep the enum and just change the signature of this `run()` method to take in those two filenames separately

Reply:
Here is the change: https://github.com/airbytehq/airbyte/pull/7501/files/59ad7e63d4b46e08119c81012ede079e6df9b054..81d33ee538f5b6f9561e9c446f8b09e2f6408564
not as pretty as before, but at least cloud can properly use this now

Reply:
Arghhh, ignore all of my comments above. I was just looking at the wrong `resources` folder in cloud, the file names actually do match between both. Going to revert this change.